test/functional/Java14andUp/src/org/openj9/test/java/lang/Test_Thread.java

```diff
@@ -0,0 +1,203 @@
+        };
+
+        /* run thread */
+        t.start();
+        AssertJUnit.assertFalse(name + "thread not yet interrupted during run", t.isInterrupted());
+        t.interrupt();
+        t.join();
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should be dead", t.isAlive());
+        AssertJUnit.assertFalse(name + "interrupt flag should not be set", t.isInterrupted()); 
+    }
+
+    /* Interrupt thread at start and during run, verify that interrupt flag is not set when thread stops running.
+     * isInterrupted becomes false after InterruptedException is thrown. */
+    @Test
+    public void test_interruptBeforeAndDuringRun() throws Throwable {
+        String name = "interruptBeforeAndDuringRun: ";
+        Thread t = new Thread() {
+            public void run() {
+                synchronized(this) {
+                    try {
+                        wait();
+                        AssertJUnit.fail(); /* not interrupted */
+                    } catch (InterruptedException e) {
+                        /* expected */
+                    }
+                }
+            }
+        };
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should not be alive", t.isAlive());
+        AssertJUnit.assertFalse(name + "interrupt flag should not be set", t.isInterrupted());
+
+        /* interrupt before run */
+        t.interrupt();
+
+        AssertJUnit.assertTrue(name + "thread that has not started should have interrupt flag set", t.isInterrupted());
+
+        /* run thread */
+        t.start();
+        t.interrupt();
+        t.join();
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should be dead", t.isAlive());
+        AssertJUnit.assertFalse(name + "interrupt flag should be cleared", t.isInterrupted()); 
+    }
+
+    /* If thread is interrupted before start, Thread.interrupted is set */
+    @Test
+    public void test_interruptedStart() throws Throwable {
+        String name = "interruptedStart: ";
+        Thread t = new Thread(){
+            public void run() throws AssertionError {
+                AssertJUnit.assertTrue(Thread.interrupted());
+            }
+          };
+
+        /* interrupt before start */
+        t.interrupt();
+
+        t.start();
+        t.join();
+
+        AssertJUnit.assertFalse(name + "interrupt flag should be cleared", t.isInterrupted()); 
+    }
+
+    /* If thread is interrupted before run, Thread.interrupted is not set */
+    @Test
+    public void test_interruptedRun() throws Throwable {
+        String name = "interrupted";
+        Thread t = new Thread(){
+            public void run() throws AssertionError {
+                AssertJUnit.assertFalse(Thread.interrupted());
+            }
+            };
+
+        /* interrupt before start */
+        t.interrupt();
+
+        t.run();
```

Comment:
This and join() don't do anything, this test is the same as test_interruptAtStartSetAfterRun()